extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -99,9 +103,11 @@ function BaseCarouselWithRef(
     [next, prev, setRestingIndex]
   );
 
-  useLayoutEffect(() => {
-    setSlideCount(length);
-  }, [setSlideCount, length]);
+  useEffect(() => {
+    if (!_thumbnails) {
```

Comment:
Then they should still not change the slides. For example, if we have:
```jsx
<InlineGallery>
  <BaseCarousel>
    <img src="img1.jpg" />
    <img src="img2.jpg" />
    <img src="img2.jpg" />
  </BaseCarousel>
  <Thumbnails />                          // Should show smaller versions of images
  <Thumbnails>                            // Should show text
    <div> img1 description </div>
    <div> img2 description </div>
    <div> img3 description </div>
  </Thumbnails/>
</InlineGallery>
```

In the case of the second `Thumbnails` instance, it provides its own slides to use but should not change the context, otherwise the first `Thumbnails` instance would display its `<div>` elements instead of the `BaseCarousel`'s `<img>` elements.

Reply:
Got it.